adding filters parameter into stock inventories report

jeremielodi opened the pull request:
This PR helps to add filter parameters into the stock inventories report, in order to discribe the result

Commit:
adding filters paramter into stock inventories report

Commit:
next part with stock report filters labels

jniles commented on server/controllers/stock/reports/index.js:
```diff
@@ -592,9 +592,29 @@ function stockLotsReport(req, res, next) {
+
+    const keys = Object.keys(defaultParam);
+    keys.forEach((key) => {
+      if ((key !== 'lang') && (key !== 'renderer')) {
```
Hmm.... this isn't really extensible.  Also, this method doesn't allow keys to be translated.

Why don't you try using something like like the `formatFilters()` function from the patient registration report?  It lists every key and it's translation value ... like this:
```js
// translation key mappings for dynamic filters
// Basically, to show a pretty filter bar, this will translate URL query params
// into human-readable text to be placed in the report, showing the properties
// filtered on.
function formatFilters(qs) {
  const columns = [
    { field : 'name', displayName : 'FORM.LABELS.NAME' },
    { field : 'sex', displayName : 'FORM.LABELS.GENDER' },
    { field : 'hospital_no', displayName : 'FORM.LABELS.HOSPITAL_NO' },
    { field : 'reference', displayName : 'FORM.LABELS.REFERENCE' },
    { field : 'dateBirthFrom', displayName : 'FORM.LABELS.DOB', comparitor : '>', isDate : true },
    { field : 'dateBirthTo', displayName : 'FORM.LABELS.DOB', comparitor : '<', isDate : true },
    { field : 'dateRegistrationFrom', displayName : 'FORM.LABELS.DATE_REGISTRATION', comparitor : '>', isDate : true },
    { field : 'dateRegistrationTo', displayName : 'FORM.LABELS.DATE_REGISTRATION', comparitor : '<', isDate : true },
    { field : 'debtor_group_uuid', displayName : 'FORM.LABELS.DEBTOR_GROUP' },
    { field : 'patient_group_uuid', displayName : 'PATIENT_GROUP.PATIENT_GROUP' },
    { field : 'user_id', displayName : 'FORM.LABELS.USER' },
    { field : 'limit', displayName : 'FORM.LABELS.LIMIT' },
    { field : 'period', displayName : 'TABLE.COLUMNS.PERIOD', isPeriod : true },
  ];

  return columns.filter(column => {

    const value = qs[column.field];

    if (!_.isUndefined(value)) {
  
      if (column.isPeriod) {
        const service = new PeriodService(new Date());
        column.value = service.periods[value].translateKey;
      } else {
        column.value = value;
      }
      return true;
    }
    return false;
  });
} 
```

This format has the following advantages:
 1. It's clear which filters are supported and which are not.
 2. The display name can be anything found in the translation dictionary.
 3. No need to `try/catch` since the code will ignore anything that isn't in the supported filters object.  
 4. Returns an array as required.
 5. Allows any additional properties to be added to a given filter (like the `isDate` property or the `comparitor` property).

jniles commented on server/controllers/stock/reports/index.js:
```diff
@@ -592,9 +592,29 @@ function stockLotsReport(req, res, next) {
+    // filters paramter found
+    hasFilter = filters.length > 0;
+
+  } catch (error) {
```
Errors should almost always be handled in some way in production.  There has to be a really good reason to have a blank `catch`.  At minimum, we should log something using `winston.error()` to let the application developers know that something when wrong.

Commit:
adding more filters in other reports

Commit:
removing old filter parameters

jniles commented on server/controllers/stock/reports/index.js:
```diff
@@ -677,17 +720,25 @@ function stockMovementsReport(req, res, next) {
     .then((rows) => {
       data.rows = rows;
       data.hasFilter = hasFilter;
+      data.filters = filters;
```
This code is no longer necessary.  Please remove it.

jniles commented on server/controllers/stock/reports/index.js:
```diff
@@ -677,17 +720,25 @@ function stockMovementsReport(req, res, next) {
       hasFilter = Object.keys(display).length > 0;
     }
 
+    filters = Object.keys(display).map(function (key) {
```
This code is no longer necessary.  Please remove it.

jniles commented on server/controllers/stock/reports/index.js:
```diff
@@ -592,9 +629,11 @@ function stockLotsReport(req, res, next) {
   let options = {};
   let display = {};
   let hasFilter = false;
+  let filters = [];
```
Do you need this array?  It looks like it is never used.

jniles reviewed (changes requested):
It looks like there is still some legacy code in here that should be removed.  After that, LGTM.

jniles committed:
chore(reports): lint stock reports

This commit fixes lint issues in the stock reports.  It also completes
the refactoring for stock filters.
